src/main/react-front-end/src/views/NavbarComponent.js

```diff
@@ -36,23 +38,34 @@ class NavbarComponent extends Component {
 	componentWillUnmount() {
 		this.unsubscribe();
 	}
+	logout(){
+		let viewPreference = cookies.get('compact') == 'true';
+		let email = store.getState().email;
+		updateViewPreference(email, viewPreference,
```

Comment:
It would be better if it is called each time the user updates it. This operation would be frequent and calling the API on every logout is unnecessary.